float64[] cx
float64[] cy